vars/cloudFoundryDeploy.groovy

```diff
@@ -251,14 +252,30 @@ def deployCfNative (config) {
             }
         }
 
+        def cfTraceFile = 'cf.log'
+
         def returnCode = sh returnStatus: true, script: """#!/bin/bash
             set +x
             set -e
             export HOME=${config.dockerWorkspace}
+            export CF_TRACE=${cfTraceFile}
             cf login -u \"${username}\" -p '${password}' -a ${config.cloudFoundry.apiEndpoint} -o \"${config.cloudFoundry.org}\" -s \"${config.cloudFoundry.space}\"
             cf plugins
             cf ${deployCommand} ${config.cloudFoundry.appName ?: ''} ${blueGreenDeployOptions} -f '${config.cloudFoundry.manifest}' ${config.smokeTest}
             """
+
+        if(config.verbose || returnCode != 0) {
+            if(fileExists(file: cfTraceFile)) {
+                echo  '### START OF CF CLI TRACE OUTPUT ###'
+                // Would be nice to inline the two next lines, but that is not understood by the test framework
```

Comment:
What about:
```
def cfTrace =  readFile(file: cfTraceFile)
echo  "### START OF CF CLI TRACE OUTPUT ###\n${cfTrace}\n### END OF CF CLI TRACE OUTPUT ###"
```

Reply:
Matter of taste. Can also be made like you outlined it. Does that make a huge difference?

Reply:
I was just commenting because of the comment